test/shopify-cli/transform_data_structure_test.rb

```diff
@@ -0,0 +1,75 @@
+require "test_helper"
+
+module ShopifyCli
+  class TransformDataStructureTest < MiniTest::Test
+    def test_primitive_values_are_returned_as_is
+      [true, false, 1, "hello", :hello].each do |value|
+        TransformDataStructure.call(value).tap do |result|
+          assert_predicate(result, :success?)
+          assert_equal(value, result.value)
+        end
+      end
+
+      TransformDataStructure.call(nil).tap do |result|
+        assert_predicate(result, :success?)
+        assert_nil(result.value)
+      end
+    end
+
+    def test_deep_symbolize_keys_of_flat_structure
+      TransformDataStructure.new(symbolize_keys: true).call({ "a" => 1, "b" => 2 }).tap do |result|
+        assert_predicate(result, :success?)
+        assert_equal({ a: 1, b: 2 }, result.value)
+      end
+    end
+
+    def test_deep_symbolize_keys_of_hash_in_array
+      TransformDataStructure.new(symbolize_keys: true).call([{ "a" => 1, "b" => 2 }]).tap do |result|
+        assert_predicate(result, :success?)
+        assert_equal([{ a: 1, b: 2 }], result.value)
+      end
+    end
+
+    def test_deeply_nested_data_structure
+      TransformDataStructure
+        .new(symbolize_keys: true)
+        .call({ "a" => [{ "b" => 1 }], "c" => { "d" => 2 } }).tap do |result|
+          assert_predicate(result, :success?)
+          assert_equal({ a: [{ b: 1 }], c: { d: 2 } }, result.value)
+        end
+    end
+
+    def test_snake_case_conversion
+      given_input = {
+        "MessageId" => 1,
+        "TTL" => 3600,
+        "useTLS" => true,
+        "Message-Priority" => "high",
+        "Transport Protocol" => "tcp",
+      }
+
+      expected_output = {
+        message_id: 1,
+        ttl: 3600,
+        use_tls: true,
+        message_priority: "high",
+        transport_protocol: "tcp",
+      }
+
+      TransformDataStructure
+        .new(symbolize_keys: true, underscore_keys: true)
```

Comment:
Same as one of my comments above, but I think this test would still pass if you changed this to `.new(symbolize_keys: false, underscore_keys: true)`

Reply:
Initially, I had implemented this to always do symbolization rather than it being a configurable option. Thanks for catching this. I have updated by code and also introduced a test that ensure that symbolization is an explicit choice and not an accidental side effect.